src/web-server/app.py

```diff
@@ -49,35 +74,58 @@ def root():
     return app.send_static_file("./index.html")
 
 
-@app.route("/<path>")
+@app.route("/<path>/")
 def load_file(path):
-    raw = get_raw(path, True)
     filename = os.path.basename(path)
     if filename.startswith("IGNORE"):
         return None
-    if raw is None:
+
+    raw = load_shortlink_file(path)
+
+    if raw is NOT_LOGGED_IN:
+        response = redirect(url_for("login"))
+        response.set_cookie(COOKIE_SHORTLINK_REDIRECT, value=path)
```

Comment:
`COOKIE_SHORTLINK_REDIRECT` is used to inform `authorize()` that we need to redirect to a shortlink, not to the root.